Mathlib/NumberTheory/ModularForms/JacobiTheta/TwoVariable.lean

```diff
@@ -14,119 +15,501 @@ This file defines the two-variable Jacobi theta function
+    simp_rw [mul_assoc (3 * π)]
+    refine mul_le_mul_of_nonneg_left ?_ (mul_pos (by norm_num : 0 < (3 : ℝ)) pi_pos).le
+    refine mul_le_mul_of_nonneg_left ?_ (pow_nonneg (Int.cast_nonneg.mpr (abs_nonneg _)) _)
+    exact norm_jacobiTheta₂_term_le hτ le_rfl le_rfl n
+
+lemma summable_jacobiTheta₂'_term_iff (z τ : ℂ) :
+    Summable (jacobiTheta₂'_term · z τ) ↔ 0 < im τ := by
+  constructor
+  · rw [← summable_jacobiTheta₂_term_iff (z := z)]
+    refine fun h ↦ (h.norm.mul_left (2 * π)⁻¹).of_norm_bounded_eventually _  ?_
+    have : ∀ᶠ (n : ℤ) in cofinite, n ≠ 0 :=
+      Int.cofinite_eq ▸ (mem_sup.mpr ⟨eventually_ne_atBot 0, eventually_ne_atTop 0⟩)
+    filter_upwards [this] with n hn
+    rw [jacobiTheta₂'_term, norm_mul, ← mul_assoc]
+    refine le_mul_of_one_le_left (norm_nonneg _) ?_
+    simp_rw [norm_mul, norm_I, norm_real, mul_one, norm_of_nonneg pi_pos.le,
+      ← ofReal_ofNat, norm_real, norm_of_nonneg two_pos.le, ← ofReal_int_cast, norm_real,
+      Real.norm_eq_abs, ← Int.cast_abs, ← mul_assoc _ (2 * π),
+      inv_mul_cancel (mul_pos two_pos pi_pos).ne', one_mul]
+    rw [← Int.cast_one, Int.cast_le]
+    exact Int.one_le_abs hn
+  · refine fun hτ ↦ ((summable_pow_mul_jacobiTheta₂_term_bound
+      |z.im| hτ 1).mul_left (2 * π)).of_norm_bounded _ (fun n ↦ ?_)
+    rw [jacobiTheta₂'_term, norm_mul, ← mul_assoc, pow_one]
+    refine mul_le_mul (le_of_eq ?_) (norm_jacobiTheta₂_term_le hτ le_rfl le_rfl n)
+      (norm_nonneg _) (by positivity)
+    simp_rw [norm_mul, Complex.norm_eq_abs, Complex.abs_two, abs_I,
+      Complex.abs_of_nonneg pi_pos.le, abs_intCast, mul_one, Int.cast_abs]
+
+end term_bounds
+
+/-!
+## Definitions of the functions
+-/
+
+/-- The two-variable Jacobi theta function,
+`θ z τ = ∑' (n : ℤ), cexp (2 * π * I * n * z + π * I * n ^ 2 * τ)`.
+-/
+def jacobiTheta₂ (z τ : ℂ) : ℂ := ∑' n : ℤ, jacobiTheta₂_term n z τ
+
+/-- Fréchet derivative of the two-variable Jacobi theta function. -/
+def jacobiTheta₂_fderiv (z τ : ℂ) : ℂ × ℂ →L[ℂ] ℂ := ∑' n : ℤ, jacobiTheta₂_term_fderiv n z τ
+
+/-- The `z`-derivative of the Jacobi theta function,
+`θ' z τ = ∑' (n : ℤ), 2 * π * I * n * cexp (2 * π * I * n * z + π * I * n ^ 2 * τ)`.
+ -/
+def jacobiTheta₂' (z τ : ℂ) := ∑' n : ℤ, jacobiTheta₂'_term n z τ
+
+lemma hasSum_jacobiTheta₂_term (z : ℂ) {τ : ℂ} (hτ : 0 < im τ) :
+    HasSum (fun n ↦ jacobiTheta₂_term n z τ) (jacobiTheta₂ z τ) :=
+  ((summable_jacobiTheta₂_term_iff z τ).mpr hτ).hasSum
+
+lemma hasSum_jacobiTheta₂_term_fderiv (z : ℂ) {τ : ℂ} (hτ : 0 < im τ) :
+    HasSum (fun n ↦ jacobiTheta₂_term_fderiv n z τ) (jacobiTheta₂_fderiv z τ) :=
+  ((summable_jacobiTheta₂_term_fderiv_iff z τ).mpr hτ).hasSum
+
+lemma hasSum_jacobiTheta₂'_term (z : ℂ) {τ : ℂ} (hτ : 0 < im τ) :
+    HasSum (fun n ↦ jacobiTheta₂'_term n z τ) (jacobiTheta₂' z τ) :=
+  ((summable_jacobiTheta₂'_term_iff z τ).mpr hτ).hasSum
+
+lemma jacobiTheta₂_undef (z : ℂ) {τ : ℂ} (hτ : im τ ≤ 0) : jacobiTheta₂ z τ = 0 := by
```

Comment:
I think most lemmas stating junk values are named using the junk value (`..._eq_zero_of_...` or so), but it appears that there are eight occurrences of `..._undef` in similar lemmas already in Mathlib.
In my opinion, `_undef` makes some sense here, but it would be nice to have more consistency...